src/Streamly/Internal/Data/Array.hs

```diff
@@ -239,3 +240,30 @@ streamFold f arr = f (toStream arr)
 {-# INLINE getIndexUnsafe #-}
 getIndexUnsafe :: Array a -> Int -> a
 getIndexUnsafe = indexArray
+
+-- | Truncate the array at the beginning and end as long as the predicate
+-- holds true.
+strip :: (a -> Bool) -> Array a -> Array a
+strip p arr =
+    let lastIndex = length arr - 1
+        indexR = getIndexR lastIndex
+        indexL = getIndexL (min 0 lastIndex)
+    in if indexL > indexR
+       then cloneArray arr 0 0
+       else cloneArray arr indexL (indexR - indexL + 1)
```

Comment:
If there is nothing to strip or if the array is empty then we can return the original array itself.